### ISSUE TITLE
Visualize the detection results on HICO_DET

### ISSUE THREAD

**Issue by vaesl** [vt-vl-lab/iCAN#16]
Thanks for sharing your code with us! iCAN is really a fantastic job, I mean that~ 
May I ask is there any code or tool to visualize the detection results on HICO_DET? I see that there exists one tools/Visualization.ipynb to visualize the detection results on V-COCO. I have tried it and it works well.
Hope to hearing from you~ thank you very much!

**Comment by Carbord** [comment]
Hello, I got some troubles while reading the "Demo/Test on your own images" in readme.md.  In the third step and fourth step, there exists the pamameter like "demo/Object_Detection.pkl" and "demo/HOI_Detection.pkl", but I cannot find these two pkl files in demo folder. So I would like to know whether these pkl files need to be trained to attain them by myself. Hope to hearing from you, thanks a lot.

**Comment by gaochen315** [comment]
@zhaowuyao, there are five steps listed in the **Demo/Test on your own images** section. Step 1 is for setting up the external object detector. Then please put your own images in the demo folder, as stated in step 2. Then you can simply run the commands showed in step 3 and step 4.  These two commands will generate these two pkl files for you. Finally, following step 5, you can use the provided tools/Demo.ipynb to visualize the HOI results.

**Comment by gaochen315** [comment]
Hi @vaesl, I am working on the iCAN v2. Once finished, I will update the visualization code which will support HICO-DET as well.

**Comment by ZHUXUHAN** [comment]
> Hi @vaesl, I am working on the iCAN v2. Once finished, I will update the visualization code which will support HICO-DET as well.

hi, good jobs have a question, the visualization.py files is just for the vcoco datasets,i also use it for the 
hoi-det,i just train it 10 epochs, but the action results for one person is much, so the visualization image is chaotic and dense, I don't konw why, can you replay why？ 
thanks

**Comment by ZHUXUHAN** [comment]
@gaochen315 
hi, good jobs have a question, the visualization.py files is just for the vcoco datasets,i also use it for the
hoi-det,i just train it 10 epochs, but the action results for one person is much, so the visualization image is chaotic and dense, I don't konw why, can you replay why？
thanks

**Comment by gaochen315** [comment]
@ZHUXUHAN, I am not sure if I fully understand your question. Several comments:

1. Do you mean by Visualization.ipynb? Visualization.py is for showing the bbox in tensorflow.
2. You can adjust the threshold. By default, it will show all the detections.
3. I also provided my HICO detection results. You can star from there.

**Comment by zhaowuyao** [comment]
step3 and step are where?What are the command contents of steps 3 and 4? I ran all the commands in your readme.txt and didn't produce  two pkl files ("demo/Object_Detection.pkl" and "demo/HOI_Detection.pkl")in demo folder.

**Comment by zhaowuyao** [comment]
Thanks .It is my fault.I have solved.

**Comment by zhaowuyao** [comment]
I have another question.In jupyter notebook,when I run Visualization.ipynb.
NameError                                 Traceback (most recent call last)
<ipython-input-24-6610f20b8a57> in <module>()
     20 
     21 
---> 22 for ele in Detection:
     23     if (ele['image_id'] == image_id):
     24         action_count = -1

NameError: name 'Detection' is not defined

**Comment by gaochen315** [comment]
@zhaowuyao, I don't understand what you mean by "run Visualization.ipynb". You have to *execute* all the sections in the provided notebook.

**Comment by zhaowuyao** [comment]
thanks!

**Comment by ZHUXUHAN** [comment]
> @ZHUXUHAN, I am not sure if I fully understand your question. Several comments:
> 
> 1. Do you mean by Visualization.ipynb? Visualization.py is for showing the bbox in tensorflow.
> 2. You can adjust the threshold. By default, it will show all the detections.
> 3. I also provided my HICO detection results. You can start from there.



> @ZHUXUHAN, I am not sure if I fully understand your question. Several comments:
> 
> 1. Do you mean by Visualization.ipynb? Visualization.py is for showing the bbox in tensorflow.
> 2. You can adjust the threshold. By default, it will show all the detections.
> 3. I also provided my HICO detection results. You can start from there.

yes,i run the visualization.ipynb,by default,the threshold is 0.01,if the visualization.ipynb can run for the 
hico detection? I also use your model provided, but the result is also not very good,  I don't know why?

**Comment by ZHUXUHAN** [comment]
@gaochen315 I see the code in demo.ipynb the model is iCAN_ResNet50_VCOCO, what can I modify it for hico-det

**Comment by CherryBear1314** [comment]
> @zhaowuyao, there are five steps listed in the **Demo/Test on your own images** section. Step 1 is for setting up the external object detector. Then please put your own images in the demo folder, as stated in step 2. Then you can simply run the commands showed in step 3 and step 4. These two commands will generate these two pkl files for you. Finally, following step 5, you can use the provided tools/Demo.ipynb to visualize the HOI results.

hi, thanks for sharing your code. I have a problem when to run the commands 3, that is “ can't open file ../tf-faster-rcnn/tools/object.Detector.py' [Errno 2] No such file or directory”. And in the tf-faster-rcnn repository , I can't find the file of object.Detector.py. So I would like to know whether this files need to be download somewhere by myself. Hope to hear from you, thank you very much!

**Comment by gaochen315** [comment]
Hi @CherryBear1314,

Did you do the first step, i.e. 
```
chmod +x ./misc/setup_demo.sh 
./misc/setup_demo.sh
```
In line 21, it copy `misc/Object_Detector.py` to `../tf-faster-rcnn/tools/` and that's the file you are looking for.

**Comment by BestSongEver** [comment]
Thanks for sharing your code. @gaochen315 
I wonder how can i  visualize the attention maps like those in your paper? 
Could anyone help? Thanx a lot.
